File: Udemy/Sec_13_WearHello/app/src/main/java/com/example/sec_13_wearhello/MainActivity.java
package com.example.sec_13_wearhello;

import android.os.Bundle;
import android.support.wearable.activity.WearableActivity;
import android.util.Log;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends WearableActivity {

    private TextView mTextView;

    public void determineRoundOrSquare(View view){
        // Get type of watch
        if(getResources().getConfiguration().isScreenRound()){
            Log.i("Device ", " Round");
            Toast.makeText(this, "ROUND", Toast.LENGTH_SHORT).show();
        }else{
            Log.i("Device ", " Square");
            Toast.makeText(this, "SQUARE", Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTextView = (TextView) findViewById(R.id.text);

        // Enables Always-on
        setAmbientEnabled();
    }
}

File: Udemy/Sec_19_Snapchat/app/src/main/java/com/example/sec_19_snapchat/MainActivity.kt
package com.example.sec_19_snapchat

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.ktx.auth
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.ktx.Firebase

class MainActivity : AppCompatActivity() {

    var emailEditText : EditText? = null
    var passwordEditText : EditText? = null
    private lateinit var auth: FirebaseAuth

    fun goClicked(view : View){
        // If we can login the user
        auth.signInWithEmailAndPassword(emailEditText?.text.toString(), passwordEditText?.text.toString())
            .addOnCompleteListener(this) { task ->
                if (task.isSuccessful) {
                    logIn();
                    Toast.makeText(this, "Login task.isSuccessful", Toast.LENGTH_SHORT).show()
                } else {
                    auth.createUserWithEmailAndPassword(emailEditText?.text.toString(), passwordEditText?.text.toString())
                        .addOnCompleteListener(this) { task ->
                            if (task.isSuccessful) {
                                // Add to Database
                                task.result?.user?.uid?.let {
                                    FirebaseDatabase.getInstance().getReference().child("users").child(
                                        it
                                    ).child("email").setValue(
                                        emailEditText?.text.toString())
                                }

                                logIn();
                            } else {
                                Toast.makeText(this, "Login Failed. Try Again", Toast.LENGTH_SHORT).show()
                            }
                        }
                }
            }
    }

    private fun logIn(){
        // Move to Next Activity
        val intent = Intent(this, SnapsActivity::class.java)
        startActivity(intent)

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Initialize Firebase Auth
        auth = Firebase.auth

        emailEditText = findViewById(R.id.emailEditText)
        passwordEditText = findViewById(R.id.passwordEitText)

    }
}
